src/main/java/jcifs/ntlmssp/Type1Message.java

```
@@ -0,0 +1,239 @@
+                flags |= NTLMSSP_NEGOTIATE_OEM_DOMAIN_SUPPLIED;
+                domain = suppliedDomainString.toUpperCase().getBytes(getOEMEncoding());
+                size += domain.length;
+            }
+            else {
+                flags &= ( NTLMSSP_NEGOTIATE_OEM_DOMAIN_SUPPLIED ^ 0xffffffff );
+            }
+
+            byte[] workstation = new byte[0];
+            String suppliedWorkstationString = getSuppliedWorkstation();
+            if ( ( flags & NTLMSSP_NEGOTIATE_VERSION ) == 0 && suppliedWorkstationString != null && suppliedWorkstationString.length() != 0 ) {
+                flags |= NTLMSSP_NEGOTIATE_OEM_WORKSTATION_SUPPLIED;
+                workstation = suppliedWorkstationString.toUpperCase().getBytes(getOEMEncoding());
+                size += workstation.length;
+            }
+            else {
+                flags &= ( NTLMSSP_NEGOTIATE_OEM_WORKSTATION_SUPPLIED ^ 0xffffffff );
+            }
+
+            byte[] type1 = new byte[size];
+            int pos = 0;
+
+            System.arraycopy(NTLMSSP_SIGNATURE, 0, type1, 0, NTLMSSP_SIGNATURE.length);
+            pos += NTLMSSP_SIGNATURE.length;
+
+            writeULong(type1, pos, NTLMSSP_TYPE1);
+            pos += 4;
+
+            writeULong(type1, pos, flags);
+            pos += 4;
+
+            int domOffOff = writeSecurityBuffer(type1, pos, domain);
+            pos += 8;
+
+            int wsOffOff = writeSecurityBuffer(type1, pos, workstation);
+            pos += 8;
+
+            if ( ( flags & NTLMSSP_NEGOTIATE_VERSION ) != 0 ) {
+                System.arraycopy(NTLMSSP_VERSION, 0, type1, pos, NTLMSSP_VERSION.length);
+                pos += NTLMSSP_VERSION.length;
+            }
+
+            pos += writeSecurityBufferContent(type1, pos, domOffOff, domain);
+            pos += writeSecurityBufferContent(type1, pos, wsOffOff, workstation);
+            return type1;
+        }
+        catch ( IOException ex ) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)